drivers/led_strip/Kconfig.ws2812

```diff
@@ -7,93 +9,33 @@
 
 menuconfig WS2812_STRIP
 	bool "Enable WS2812 (and compatible) LED strip driver"
-	depends on SPI
+	select LED_STRIP_RGB_SCRATCH
 	help
 	  Enable LED strip driver for daisy chains of WS2812-ish
 	  (or WS2812B, WS2813, SK6812, or compatible) devices.
-	  These devices have a one-wire communications interface
-	  which encodes bits using pulses. Short pulses indicate
-	  zero bits, and long pulses indicate ones; refer to the
-	  chip datasheets for precise specifications. To implement
-	  this in a multitasking operating system, this driver
-	  generates the pulses using a SPI peripheral.
 
 if WS2812_STRIP
 
-config WS2812_STRIP_MAX_PIXELS
-	int "Maximum number of pixels in a strip"
-	default 12
-	help
-	  Set this to the maximum number of pixels you need
-	  to control at once. There is an 8x memory penalty associated
-	  with each increment of this value, so it's worth optimizing.
-
-config WS2812_STRIP_ONE_FRAME
-	hex "SPI frame to shift out to signal a one bit"
-	default 0x7c if SOC_SERIES_STM32F4X
-	default 0x70 if SOC_FAMILY_NRF
-	help
-	  When shifted out at the configured clock frequency,
-	  this must generate a pulse whose width fits within the chipset
-	  specifications for T1H, and whose interpulse timing meets low
-	  times. It is recommended that the first and last bits in the
-	  frame be zero; this "encourages" SPI IPs to leave MOSI low
-	  between frames.
-
-config WS2812_STRIP_ZERO_FRAME
-	hex "SPI frame to shift out to signal a zero bit"
-	default 0x60 if SOC_SERIES_STM32F4X
-	default 0x40 if SOC_FAMILY_NRF
-	help
-	  When shifted out at the configured clock frequency,
-	  this must generate a pulse whose width fits within the chipset
-	  specifications for T0H, and whose interpulse timing meets low
-	  times. It is recommended that the first and last bits in the
-	  frame be zero; this "encourages" SPI IPs to leave MOSI low
-	  between frames.
-
-# By default, we use GRBW [sic] (and ignore W).
-comment "The following options determine channel data order on the wire."
-
-config WS2812_RED_ORDER
-	int "Order in which a red pixel should be shifted out"
-	default 1
-	range 0 3
-	help
-	  If the red channel is shifted out first, specify 0.
-	  If second, specify 1, and so on.
+choice WS2812_STRIP_DRIVER
```

Comment:
Since we allow for multiple instances maybe we should also allow to have a mix, e.g. one instance of GPIO and one instance of SPI based driver. In this case we shouldn't have a choice but two independent options.

Reply:
I think the changes I've made in this PR have already gotten quite a bit larger than the original request, which was to update it to the new GPIO API only, so I hope you're OK with me waiting to do this until someone actually needs it.

In my opinion, having multiple strips is already a bit of a stretch. Practically speaking, an application using this hardware is going to have one logical strip, since the device hardware is a daisy chain.

I can't think of a practical situation in which I would design an embedded device which would want both drivers enabled in software.

Reply:
Sure, that was a major rework and the code looked good from the very beginning.